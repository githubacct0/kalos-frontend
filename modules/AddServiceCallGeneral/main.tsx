import React, { FC, useCallback, useEffect, useState } from 'react';
import { AddServiceCall, Props } from './components/AddServiceCall';
import { PageWrapper, PageWrapperProps } from '../PageWrapper/main';
import { Tabs } from '../ComponentsLibrary/Tabs';
import { ServiceCallSearch } from '../ServiceCallSearch/main';
import { AdvancedSearch } from '../ComponentsLibrary/AdvancedSearch';
import { GanttChart } from '../ComponentsLibrary/GanttChart';
import { EventType, loadProjects } from '../../helpers';
import { Loader } from '../Loader/main';
import { Confirm } from '../ComponentsLibrary/Confirm';

export const AddServiceCallGeneral: FC<Props & PageWrapperProps> = props => {
  const [projects, setProjects] = useState<EventType[]>([]);
  const [loading, setLoading] = useState<boolean>(false);
  const [loaded, setLoaded] = useState<boolean>(false);
  const [parentId, setParentId] = useState<number | null>(null);
  const [confirmedParentId, setConfirmedParentId] = useState<number | null>(
    null,
  );

  const handleSetParentId = useCallback(
    id => {
      console.log('Set parent id: ', id);
      setParentId(id);
    },
    [setParentId],
  );

  const handleSetConfirmedIsChild = useCallback(
    id => {
      console.log('Confirmed is child: ', id);
      //setConfirmedIsChild(confirmed);
      setConfirmedParentId(id);
    },
    [setConfirmedParentId],
  );

  const load = useCallback(async () => {
    setLoading(true);
    const projects = await loadProjects();
    console.log('Got projects: ', projects);
    setProjects(projects);
    setLoading(false);
    setLoaded(true);
  }, [setProjects, setLoading, setLoaded]);

  useEffect(() => {
    load();
  }, [load, setProjects, setLoading, setLoaded]);

  return (
    <PageWrapper {...props} userID={props.loggedUserId}>
      <>
        <Tabs
          tabs={Array.from(Array(2)).map((_, idx) => ({
            label: idx == 0 ? `Add Service Call` : `Add Project`,
            content: (
              <AddServiceCall {...props} asProject={idx == 0 ? false : true} />
            ),
          }))}
        />
        {parentId != confirmedParentId && parentId != null && (
          <Confirm
            title="Confirm Parent"
            open={true}
            onClose={() => handleSetParentId(null)}
            onConfirm={() => handleSetConfirmedIsChild(parentId)}
          >
            Are you sure you want to set this project as the parent to the new
            project?
          </Confirm>
        )}
        {loaded ? (
          <GanttChart
            events={projects.map(task => {
              const {
                id,
                description,
                dateStarted: dateStart,
                dateEnded: dateEnd,
                logJobStatus,
                color,
              } = task;
              const [startDate, startHour] = dateStart.split(' ');
              const [endDate, endHour] = dateEnd.split(' ');
              return {
                id,
                startDate,
                endDate,
                startHour,
                endHour,
                notes: description,
                statusColor: '#' + color,
                onClick: () => {
                  console.log('Service call id of clicked: ', id);
                  handleSetParentId(id);
                  // setServiceCallId(id);
                  // setLoaded(false);
                  // setLoadedInit(false);
                  // // load();
                },
              };
            })}
            startDate={projects[0].dateStarted.substr(0, 10)}
            endDate={projects[projects.length - 1].dateEnded.substr(0, 10)}
            loading={loading}
          />
        ) : (
          <Loader />
        )}
      </>
    </PageWrapper>
  );
};
