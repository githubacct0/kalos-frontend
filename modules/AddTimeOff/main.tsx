import React, { FC } from 'react';
import { PageWrapper, PageWrapperProps } from '../PageWrapper/main';
import { TimeOff, Props as TimeOffProps } from '../ComponentsLibrary/TimeOff';

export const AddTimeOff: FC<TimeOffProps & PageWrapperProps> = props => (
  <PageWrapper {...props} userID={props.loggedUserId} withHeader>
    <TimeOff {...props} />
  </PageWrapper>
);
