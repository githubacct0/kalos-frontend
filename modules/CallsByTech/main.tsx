import React from 'react';
import IconButton from '@material-ui/core/IconButton';
import EditIcon from '@material-ui/icons/Edit';
import {
  formatTime,
  makeFakeRows,
  loadEventsByFilter,
  UserClientService,
} from '../../helpers';
import { InfoTable } from '../ComponentsLibrary/InfoTable';
import { Modal } from '../ComponentsLibrary/Modal';
import { ServiceCall } from '../ComponentsLibrary/ServiceCall';
import { Form, Schema } from '../ComponentsLibrary/Form';
import { PageWrapper, PageWrapperProps } from '../PageWrapper/main';
import { getPropertyAddress } from '../../@kalos-core/kalos-rpc/Property';
import { Event as EventType } from '../../@kalos-core/kalos-rpc/Event';

interface Props extends PageWrapperProps {
  userId: number;
}

type SearchForm = {
  employeeId: string;
  date: string;
};

interface state {
  calls: EventType[];
  isLoading: boolean;
  editing?: EventType;
  searchForm: SearchForm;
}

const SCHEMA_SEARCH: Schema<SearchForm> = [
  [
    {
      name: 'employeeId',
      label: 'Employee',
      type: 'technician',
      required: true,
    },
    {
      name: 'date',
      label: 'Date Started',
      type: 'date',
      required: true,
    },
  ],
];

export class CallsByTech extends React.PureComponent<Props, state> {
  constructor(props: Props) {
    super(props);
    this.state = {
      calls: [],
      isLoading: false,
      searchForm: {
        employeeId: '',
        date: '',
      },
    };
  }

  toggleLoading = (): Promise<boolean> =>
    new Promise(resolve => {
      this.setState(
        prevState => ({
          isLoading: !prevState.isLoading,
        }),
        () => resolve(true),
      );
    });

  toggleEditing = (editing?: EventType) => () => this.setState({ editing });

  search = (searchForm: SearchForm) =>
    this.setState({ searchForm }, this.loadEvents);

  loadEvents = async () => {
    await this.toggleLoading();
    let newEvent = new EventType();
    const { resultsList: calls } = await loadEventsByFilter({
      page: -1,
      filter: {
        dateStarted: `${this.state.searchForm.date} 00:00:00`,
        logTechnicianAssigned: `%${this.state.searchForm.employeeId}%`,
      },
      sort: {
        orderBy: 'time_started',
        orderByField: 'getTimeStarted',
        orderDir: 'ASC',
      },
      req: newEvent,
    });
    this.setState({ calls });
    await this.toggleLoading();
  };

  async componentDidMount() {}

  render() {
    const { editing, isLoading } = this.state;
    return (
      <PageWrapper {...this.props} userID={this.props.userId}>
        <Form<SearchForm>
          title="Service Calls by Employee"
          schema={SCHEMA_SEARCH}
          data={{
            employeeId: '',
            date: new Date().toISOString().substr(0, 10),
          }}
          onSave={this.search}
          submitLabel="Search"
          onClose={null}
        />
        <InfoTable
          loading={isLoading}
          columns={[
            { name: 'Time' },
            { name: 'Customer' },
            { name: 'Address' },
            { name: 'Phone' },
            { name: 'Job #' },
            { name: 'Job Type' },
            { name: 'Job Status' },
          ]}
          data={
            isLoading
              ? makeFakeRows(7, 5)
              : this.state.calls.map(c => [
                  {
                    value: `${formatTime(c.getTimeStarted())} - ${formatTime(
                      c.getTimeEnded(),
                    )}`,
                    onClick: this.toggleEditing(c),
                  },
                  {
                    value: UserClientService.getCustomerNameAndBusinessName(
                      c.getCustomer()!,
                    ),
                    onClick: this.toggleEditing(c),
                  },
                  {
                    value: getPropertyAddress(c.getProperty()),
                    onClick: this.toggleEditing(c),
                  },
                  {
                    value: UserClientService.getCustomerPhone(c.getCustomer()!),
                    onClick: this.toggleEditing(c),
                  },
                  {
                    value: c.getId(),
                    onClick: this.toggleEditing(c),
                  },
                  {
                    value: `${c.getJobType()} / ${c.getJobSubtype()}`,
                    onClick: this.toggleEditing(c),
                  },
                  {
                    value: c.getLogJobStatus(),
                    onClick: this.toggleEditing(c),
                    actions: c.getCustomer()
                      ? [
                          <IconButton
                            key="edit"
                            size="small"
                            onClick={this.toggleEditing(c)}
                          >
                            <EditIcon />
                          </IconButton>,
                        ]
                      : [],
                  },
                ])
          }
        />
        {editing && editing.getCustomer() && (
          <Modal open onClose={this.toggleEditing()} fullScreen>
            <ServiceCall
              loggedUserId={this.props.userId}
              propertyId={editing.getPropertyId()}
              userID={editing.getCustomer()!.getId()}
              serviceCallId={editing.getId()}
              onClose={this.toggleEditing()}
            />
          </Modal>
        )}
      </PageWrapper>
    );
  }
}
