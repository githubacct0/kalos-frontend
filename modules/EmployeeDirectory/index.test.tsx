import React from 'react';
import renderer from 'react-test-renderer';
import { EmployeeDirectory } from './main';

test('EmployeeDirectory renders correctly', () => {
  const tree = renderer
    .create(<EmployeeDirectory loggedUserId={101253} withHeader />)
    .toJSON();
  expect(tree).toMatchSnapshot(); // Rendered correctly
});
