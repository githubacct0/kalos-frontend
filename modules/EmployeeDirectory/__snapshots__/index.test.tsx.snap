// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployeeDirectory renders correctly 1`] = `
<div
  className="PageWrapper"
>
  <div
    className="PageWrapperMenu"
  />
  <div
    className="PageWrapperContent"
    style={
      Object {
        "padding": "0rem",
      }
    }
  >
    <div>
      <div
        className="SectionBar fixedActions sticky"
      >
        <div
          className="SectionBarHeaderWrapper"
        >
          <div
            className="SectionBarHeader"
          >
            <div
              className="SectionBarTitleWrapper"
            >
              <h5
                className="MuiTypography-root SectionBarTitle MuiTypography-h5"
                onClick={[Function]}
              >
                Employee Directory
                 
              </h5>
            </div>
          </div>
          <div
            className="SectionBarActions"
          >
            <div
              className="Actions"
            >
              <div
                className="ActionsActions"
              >
                <button
                  className="MuiButtonBase-root MuiButton-root MuiButton-contained ButtonWrapper ActionsButton size-small status-undefined variant-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiButton-label"
                  >
                    Employee Departments
                  </span>
                </button>
              </div>
            </div>
            <span
              className=""
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-contained ButtonWrapper size-small status-undefined variant-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  Print
                </span>
              </button>
            </span>
            <div
              className="PrintPage"
            >
              <div>
                <div
                  className="PrintHeader"
                >
                  <div
                    className="PrintHeader_logo"
                  />
                  <div
                    className="PrintHeader_content"
                  >
                    <div
                      className="PrintParagraph h1"
                      style={
                        Object {
                          "marginTop": 0,
                          "textAlign": "center",
                        }
                      }
                    >
                      Employees
                    </div>
                    <div
                      className="PrintParagraph div"
                      style={
                        Object {
                          "textAlign": "center",
                        }
                      }
                    >
                       
                    </div>
                  </div>
                  <div />
                </div>
                <table
                  className="PrintPage_table"
                >
                  <tbody>
                    <tr>
                      <td>
                        <table
                          className="PrintTable"
                          style={Object {}}
                        >
                          <thead>
                            <tr>
                              <th
                                style={Object {}}
                              >
                                Name
                              </th>
                              <th
                                style={Object {}}
                              >
                                Title
                              </th>
                              <th
                                style={Object {}}
                              >
                                Email
                              </th>
                              <th
                                style={Object {}}
                              >
                                Office, ext.
                              </th>
                              <th
                                style={
                                  Object {
                                    "textAlign": "right",
                                  }
                                }
                              >
                                Cell
                              </th>
                            </tr>
                          </thead>
                          <tbody>
                            <tr>
                              <td
                                style={Object {}}
                              >
                                
                              </td>
                              <td
                                style={Object {}}
                              >
                                
                              </td>
                              <td
                                style={Object {}}
                              >
                                
                              </td>
                              <td
                                style={Object {}}
                              >
                                
                              </td>
                              <td
                                style={
                                  Object {
                                    "textAlign": "right",
                                  }
                                }
                              >
                                
                              </td>
                            </tr>
                            <tr>
                              <td
                                style={Object {}}
                              >
                                
                              </td>
                              <td
                                style={Object {}}
                              >
                                
                              </td>
                              <td
                                style={Object {}}
                              >
                                
                              </td>
                              <td
                                style={Object {}}
                              >
                                
                              </td>
                              <td
                                style={
                                  Object {
                                    "textAlign": "right",
                                  }
                                }
                              >
                                
                              </td>
                            </tr>
                            <tr>
                              <td
                                style={Object {}}
                              >
                                
                              </td>
                              <td
                                style={Object {}}
                              >
                                
                              </td>
                              <td
                                style={Object {}}
                              >
                                
                              </td>
                              <td
                                style={Object {}}
                              >
                                
                              </td>
                              <td
                                style={
                                  Object {
                                    "textAlign": "right",
                                  }
                                }
                              >
                                
                              </td>
                            </tr>
                          </tbody>
                        </table>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
      <form
        className="AdvancedSearchForm PlainForm compact"
        onSubmit={[Function]}
      >
        <div
          className="PlainFormGroup"
        >
          <div
            className="Field type-search"
            style={Object {}}
          >
            <div
              className="MuiFormControl-root MuiTextField-root FieldInput MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
                data-shrink={true}
              >
                First Name
                
              </label>
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiInput-input MuiInputBase-inputTypeSearch MuiInput-inputTypeSearch"
                  disabled={false}
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  readOnly={false}
                  required={false}
                  type="search"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="Field PlainFormField type-search"
            style={Object {}}
          >
            <div
              className="MuiFormControl-root MuiTextField-root FieldInput MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
                data-shrink={true}
              >
                Last Name
                
              </label>
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiInput-input MuiInputBase-inputTypeSearch MuiInput-inputTypeSearch"
                  disabled={false}
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  readOnly={false}
                  required={false}
                  type="search"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="Field PlainFormField"
            style={Object {}}
          >
            <div
              className="MuiFormControl-root FieldInput MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                data-shrink={true}
                id="employeeDepartmentId-select-label"
              >
                Department
                
              </label>
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                onClick={[Function]}
              >
                <div
                  aria-haspopup="listbox"
                  aria-labelledby="employeeDepartmentId-select-label employeeDepartmentId-select"
                  className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                  id="employeeDepartmentId-select"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onMouseDown={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  -- All --
                </div>
                <input
                  aria-hidden={true}
                  className="MuiSelect-nativeInput"
                  onAnimationStart={[Function]}
                  onChange={[Function]}
                  required={false}
                  tabIndex={-1}
                  value={-1}
                />
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSelect-icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
               
            </div>
          </div>
          <div
            className="Field PlainFormField type-search"
            style={Object {}}
          >
            <div
              className="MuiFormControl-root MuiTextField-root FieldInput MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
                data-shrink={true}
              >
                Title
                
              </label>
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiInput-input MuiInputBase-inputTypeSearch MuiInput-inputTypeSearch"
                  disabled={false}
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  readOnly={false}
                  required={false}
                  type="search"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="PlainFormGroup"
        >
          <div
            className="Field type-search"
            style={Object {}}
          >
            <div
              className="MuiFormControl-root MuiTextField-root FieldInput MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
                data-shrink={true}
              >
                Email
                
              </label>
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiInput-input MuiInputBase-inputTypeSearch MuiInput-inputTypeSearch"
                  disabled={false}
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  readOnly={false}
                  required={false}
                  type="search"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="Field PlainFormField type-search"
            style={Object {}}
          >
            <div
              className="MuiFormControl-root MuiTextField-root FieldInput MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
                data-shrink={true}
              >
                Cell
                
              </label>
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiInput-input MuiInputBase-inputTypeSearch MuiInput-inputTypeSearch"
                  disabled={false}
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  readOnly={false}
                  required={false}
                  type="search"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="Field PlainFormField type-search"
            style={Object {}}
          >
            <div
              className="MuiFormControl-root MuiTextField-root FieldInput MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
                data-shrink={true}
              >
                Office
                
              </label>
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiInput-input MuiInputBase-inputTypeSearch MuiInput-inputTypeSearch"
                  disabled={false}
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  readOnly={false}
                  required={false}
                  type="search"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="Field PlainFormField type-search"
            style={Object {}}
          >
            <div
              actions={
                Array [
                  Object {
                    "label": "Reset",
                    "onClick": [Function],
                    "variant": "outlined",
                  },
                ]
              }
              className="MuiFormControl-root MuiTextField-root FieldInput MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
                data-shrink={true}
              >
                Ext.
                
              </label>
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiInput-input MuiInputBase-inputTypeSearch MuiInput-inputTypeSearch"
                  disabled={false}
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  readOnly={false}
                  required={false}
                  type="search"
                  value=""
                />
              </div>
            </div>
            <div
              className="Actions FieldActions"
            >
              <div
                className="ActionsActions responsiveColumn"
              >
                <button
                  className="MuiButtonBase-root MuiButton-root MuiButton-outlined ButtonWrapper ActionsButton responsiveColumn size-small status-undefined variant-outlined MuiButton-outlinedPrimary MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiButton-label"
                  >
                    Reset
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
        <button
          className="PlainFormSubmit"
          type="submit"
        >
          Submit
        </button>
      </form>
      <div
        className="InfoTable"
      >
        <div
          className="InfoTableHeader"
        >
          <div
            className="MuiTypography-root InfoTableColumn MuiTypography-body1"
            style={
              Object {
                "flexGrow": 0,
                "flexShrink": 1,
                "width": "20%",
              }
            }
          >
            <span
              className="InfoTableDir"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                  "justifyContent": "flex-start",
                }
              }
            >
              Name
               
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 14l5-5 5 5z"
                />
              </svg>
            </span>
          </div>
          <div
            className="MuiTypography-root InfoTableColumn MuiTypography-body1"
            style={
              Object {
                "flexGrow": 0,
                "flexShrink": 1,
                "width": "20%",
              }
            }
          >
            <span
              className="InfoTableDir"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                  "justifyContent": "flex-start",
                }
              }
            >
              Title
               
            </span>
          </div>
          <div
            className="MuiTypography-root InfoTableColumn MuiTypography-body1"
            style={
              Object {
                "flexGrow": 0,
                "flexShrink": 1,
                "width": "20%",
              }
            }
          >
            <span
              className="InfoTableDir"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                  "justifyContent": "flex-start",
                }
              }
            >
              Email
               
            </span>
          </div>
          <div
            className="MuiTypography-root InfoTableColumn MuiTypography-body1"
            style={
              Object {
                "flexGrow": 0,
                "flexShrink": 1,
                "width": "20%",
              }
            }
          >
            <span
              className="InfoTableDir"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                  "justifyContent": "flex-start",
                }
              }
            >
              Office, ext.
               
            </span>
          </div>
          <div
            className="MuiTypography-root InfoTableColumn MuiTypography-body1"
            style={
              Object {
                "flexGrow": 0,
                "flexShrink": 1,
                "width": "20%",
              }
            }
          >
            <span
              className="InfoTableDir"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                  "justifyContent": "flex-start",
                }
              }
            >
              Cell
               
            </span>
          </div>
        </div>
        <div
          className="InfoTableRow hoverable"
        >
          <div
            className="MuiTypography-root InfoTableItem MuiTypography-body1"
            style={
              Object {
                "cursor": "default",
                "flexGrow": 0,
                "flexShrink": 1,
                "justifyContent": "flex-start",
                "width": "20%",
              }
            }
          >
            <span
              className="InfoTableFake"
            />
          </div>
          <div
            className="MuiTypography-root InfoTableItem MuiTypography-body1"
            style={
              Object {
                "cursor": "default",
                "flexGrow": 0,
                "flexShrink": 1,
                "justifyContent": "flex-start",
                "width": "20%",
              }
            }
          >
            <span
              className="InfoTableFake"
            />
          </div>
          <div
            className="MuiTypography-root InfoTableItem MuiTypography-body1"
            style={
              Object {
                "cursor": "default",
                "flexGrow": 0,
                "flexShrink": 1,
                "justifyContent": "flex-start",
                "width": "20%",
              }
            }
          >
            <span
              className="InfoTableFake"
            />
          </div>
          <div
            className="MuiTypography-root InfoTableItem MuiTypography-body1"
            style={
              Object {
                "cursor": "default",
                "flexGrow": 0,
                "flexShrink": 1,
                "justifyContent": "flex-start",
                "width": "20%",
              }
            }
          >
            <span
              className="InfoTableFake"
            />
          </div>
          <div
            className="MuiTypography-root InfoTableItem MuiTypography-body1"
            style={
              Object {
                "cursor": "default",
                "flexGrow": 0,
                "flexShrink": 1,
                "justifyContent": "flex-start",
                "width": "20%",
              }
            }
          >
            <span
              className="InfoTableFake"
            />
          </div>
        </div>
        <div
          className="InfoTableRow hoverable"
        >
          <div
            className="MuiTypography-root InfoTableItem MuiTypography-body1"
            style={
              Object {
                "cursor": "default",
                "flexGrow": 0,
                "flexShrink": 1,
                "justifyContent": "flex-start",
                "width": "20%",
              }
            }
          >
            <span
              className="InfoTableFake"
            />
          </div>
          <div
            className="MuiTypography-root InfoTableItem MuiTypography-body1"
            style={
              Object {
                "cursor": "default",
                "flexGrow": 0,
                "flexShrink": 1,
                "justifyContent": "flex-start",
                "width": "20%",
              }
            }
          >
            <span
              className="InfoTableFake"
            />
          </div>
          <div
            className="MuiTypography-root InfoTableItem MuiTypography-body1"
            style={
              Object {
                "cursor": "default",
                "flexGrow": 0,
                "flexShrink": 1,
                "justifyContent": "flex-start",
                "width": "20%",
              }
            }
          >
            <span
              className="InfoTableFake"
            />
          </div>
          <div
            className="MuiTypography-root InfoTableItem MuiTypography-body1"
            style={
              Object {
                "cursor": "default",
                "flexGrow": 0,
                "flexShrink": 1,
                "justifyContent": "flex-start",
                "width": "20%",
              }
            }
          >
            <span
              className="InfoTableFake"
            />
          </div>
          <div
            className="MuiTypography-root InfoTableItem MuiTypography-body1"
            style={
              Object {
                "cursor": "default",
                "flexGrow": 0,
                "flexShrink": 1,
                "justifyContent": "flex-start",
                "width": "20%",
              }
            }
          >
            <span
              className="InfoTableFake"
            />
          </div>
        </div>
        <div
          className="InfoTableRow hoverable"
        >
          <div
            className="MuiTypography-root InfoTableItem MuiTypography-body1"
            style={
              Object {
                "cursor": "default",
                "flexGrow": 0,
                "flexShrink": 1,
                "justifyContent": "flex-start",
                "width": "20%",
              }
            }
          >
            <span
              className="InfoTableFake"
            />
          </div>
          <div
            className="MuiTypography-root InfoTableItem MuiTypography-body1"
            style={
              Object {
                "cursor": "default",
                "flexGrow": 0,
                "flexShrink": 1,
                "justifyContent": "flex-start",
                "width": "20%",
              }
            }
          >
            <span
              className="InfoTableFake"
            />
          </div>
          <div
            className="MuiTypography-root InfoTableItem MuiTypography-body1"
            style={
              Object {
                "cursor": "default",
                "flexGrow": 0,
                "flexShrink": 1,
                "justifyContent": "flex-start",
                "width": "20%",
              }
            }
          >
            <span
              className="InfoTableFake"
            />
          </div>
          <div
            className="MuiTypography-root InfoTableItem MuiTypography-body1"
            style={
              Object {
                "cursor": "default",
                "flexGrow": 0,
                "flexShrink": 1,
                "justifyContent": "flex-start",
                "width": "20%",
              }
            }
          >
            <span
              className="InfoTableFake"
            />
          </div>
          <div
            className="MuiTypography-root InfoTableItem MuiTypography-body1"
            style={
              Object {
                "cursor": "default",
                "flexGrow": 0,
                "flexShrink": 1,
                "justifyContent": "flex-start",
                "width": "20%",
              }
            }
          >
            <span
              className="InfoTableFake"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
